tfx/examples/iris/iris_pipeline_native_keras_custom_splits.py

```diff
@@ -0,0 +1,241 @@
+_module_file = os.path.join(_iris_root, 'iris_utils_native_keras.py')
+# Path which can be listened to by the model server.  Pusher will output the
+# trained model here.
+_serving_model_dir = os.path.join(_iris_root, 'serving_model', _pipeline_name)
+
+# Directory and data locations.  This example assumes all of the flowers
+# example code and metadata library is relative to $HOME, but you can store
+# these files anywhere on your local filesystem.
+_tfx_root = os.path.join(os.environ['HOME'], 'tfx')
+_pipeline_root = os.path.join(_tfx_root, 'pipelines', _pipeline_name)
+# Sqlite ML-metadata db path.
+_metadata_path = os.path.join(_tfx_root, 'metadata', _pipeline_name,
+                              'metadata.db')
+
+# Pipeline arguments for Beam powered Components.
+_beam_pipeline_args = [
+    '--direct_running_mode=multi_processing',
+    # 0 means auto-detect based on on the number of CPUs available
+    # during execution time.
+    '--direct_num_workers=0',
+]
+
+
+def _create_pipeline(pipeline_name: Text, pipeline_root: Text, data_root: Text,
+                     module_file: Text, serving_model_dir: Text,
+                     metadata_path: Text, enable_tuning: bool,
+                     beam_pipeline_args: List[Text]) -> pipeline.Pipeline:
+  """Implements the Iris flowers pipeline with TFX."""
+  examples = external_input(data_root)
+
+  output = example_gen_pb2.Output(
+      split_config=example_gen_pb2.SplitConfig(splits=[
+          example_gen_pb2.SplitConfig.Split(name='training', hash_buckets=3),
+          example_gen_pb2.SplitConfig.Split(name='evaluating', hash_buckets=2),
+          example_gen_pb2.SplitConfig.Split(name='testing', hash_buckets=1)
+      ]))
+
+  # Brings data into the pipeline or otherwise joins/converts training data.
+  example_gen = CsvExampleGen(input=examples, output_config=output)
+
+  # Computes statistics over data for visualization and example validation.
+  statistics_gen = StatisticsGen(examples=example_gen.outputs['examples'],
+                                 exclude_splits=['testing'])
+
+  # Generates schema based on statistics files.
+  schema_gen = SchemaGen(
+      statistics=statistics_gen.outputs['statistics'], infer_feature_shape=True,
+      exclude_splits=['testing'])
+
+  # Performs anomaly detection based on statistics and data schema.
+  example_validator = ExampleValidator(
+      statistics=statistics_gen.outputs['statistics'],
+      schema=schema_gen.outputs['schema'],
+      exclude_splits=['testing'])
+
+  # Performs transformations and feature engineering in training and serving.
+  transform = Transform(
+      examples=example_gen.outputs['examples'],
+      schema=schema_gen.outputs['schema'],
+      module_file=module_file,
+      splits_config=transform_pb2.SplitsConfig(analyze=['training'],
+                                               transform=['training',
+                                                          'evaluating']))
```

Comment:
FYI, transform is just submitted, you can download tfx head and install tfx head to test now

Reply:
yes, I was planning to wait until transform is merged and test it again